File: api/app/routers/health.py
from fastapi import APIRouter, Depends
from fastapi.responses import JSONResponse
from sqlalchemy import text
from sqlalchemy.ext.asyncio import AsyncSession

from app.database import get_session
from app.schemas.health import HealthResponse

router = APIRouter(tags=["health"])


@router.get("/health", response_model=HealthResponse)
async def health_check(session: AsyncSession = Depends(get_session)):
    try:
        await session.execute(text("SELECT 1"))
        return HealthResponse(status="healthy", db="connected")
    except Exception:
        return JSONResponse(
            status_code=503,
            content={"status": "unhealthy", "db": "disconnected"},
        )

File: api/src/routes/health.ts
import { sql } from "drizzle-orm";
import { Hono } from "hono";
import { describeRoute } from "hono-openapi";
import type { DB } from "../database.js";
import { jsonResponse } from "../openapi.js";
import { healthResponseSchema } from "../schemas/health.js";
import type { AppEnv } from "../types.js";

export function createHealthRoute(db: DB): Hono<AppEnv> {
  const route = new Hono<AppEnv>();

  route.get(
    "/health",
    describeRoute({
      tags: ["Health"],
      summary: "Liveness and database connectivity",
      responses: {
        200: jsonResponse("Healthy", healthResponseSchema),
        503: jsonResponse("Database unreachable", healthResponseSchema),
      },
    }),
    async (c) => {
      try {
        await db.execute(sql`SELECT 1`);
        return c.json({ status: "healthy", db: "connected" });
      } catch {
        return c.json({ status: "unhealthy", db: "disconnected" }, 503);
      }
    },
  );

  return route;
}
